src/components/dialogs/renew.vue

```diff
@@ -0,0 +1,252 @@
+              <v-btn
+                @click="hideModal()"
+                id="renew-cancel-btn"
+                class="button button-blue px-5">Cancel</v-btn>
+              <v-btn
+                @click="confirmPayment()"
+                id="renew-continue-btn"
+                class="primary px-5"
+                :loading="isLoadingPayment">Continue to Payment</v-btn>
+            </v-card-actions>
+          </v-tab-item>
+
+        </v-tabs-items>
+      </v-tabs>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang='ts'>
+import { Component, Mixins, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+import { getFeatureFlag } from '@/plugins'
+import FeeSummary from '@/components/payment/fee-summary.vue'
+import StaffPayment from '@/components/payment/staff-payment.vue'
+import { CreatePaymentParams, FetchFeesParams } from '@/modules/payment/models'
+import { RENEW_MODAL_IS_VISIBLE } from '@/modules/payment/store/types'
+import { FilingTypes } from '@/modules/payment/filing-types'
+import { Jurisdictions, PaymentAction } from '@/enums'
+import { PaymentMixin, PaymentSessionMixin, DisplayedComponentMixin } from '@/mixins'
+import { getBaseUrl } from '@/components/payment/payment-utils'
+import { ActionBindingIF } from '@/interfaces/store-interfaces'
+
+@Component({
+  components: {
+    FeeSummary,
+    StaffPayment
+  }
+})
+export default class RenewDialog extends Mixins(
+  PaymentMixin,
+  PaymentSessionMixin,
+  DisplayedComponentMixin
+) {
+  // the tab indices
+  // NB: these are reversed to reverse the built-in slide transition
+  readonly TAB_STAFF_PAYMENT = 0
+  readonly TAB_RENEW_NAME_REQUEST = 1
+
+  // Refs
+  $refs!: {
+    staffPaymentComponent: StaffPayment
+  }
+
+  // Global getters
+  @Getter getName!: string
+  @Getter isRoleStaff!: boolean
+
+  // Global action
+  @Action toggleRenewModal!: ActionBindingIF
+
+  /** Whether staff payment is valid. */
+  private isStaffPaymentValid = false
+
+  /** The current tab to display. */
+  private currentTab = this.TAB_RENEW_NAME_REQUEST
+
+  /** Whether payment redirection is in progress. */
+  private isLoadingPayment = false
+
+  /** Whether this dialog is visible. */
+  private isVisible = false
+
+  /** Whether this modal should be shown (per store property). */
+  private get showModal (): boolean {
+    return this.$store.getters[RENEW_MODAL_IS_VISIBLE]
+  }
+
+  /** Clears store property to hide this modal. */
+  async hideModal () {
+    this.isLoadingPayment = false
+    await this.toggleRenewModal(false)
+  }
+
+  /** Depending on value, fetches fees and makes this modal visible or hides it. */
+  @Watch('showModal')
+  async onShowModal (val: boolean): Promise<void> {
+    if (val) {
+      // reset tab id
+      this.currentTab = this.TAB_RENEW_NAME_REQUEST
+
+      const params: FetchFeesParams = {
+        filingType: FilingTypes.NM620,
+        jurisdiction: Jurisdictions.BC,
```

Comment:
This code was cloned from the pre-existing Upgrade modal. AFAIK, yes, it's always BC. (I'm guessing this is the jurisdiction in which we are charging fees, so that PST/GST is added as needed.)

The enum for FilingTypes has a comment that NM620 is a "regular submission online". In other words, just a regular NR creation. (The only other code that we use, as confirmed by Sumesh in RC, is NM606 - "upgrade to priority".)